src/OrleansAzureUtils/Providers/Streams/PersistentStreams/StreamDeliveryFailureEntity.cs

```diff
@@ -0,0 +1,124 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System;
+using Microsoft.WindowsAzure.Storage.Table;
+using Orleans.Serialization;
+using Orleans.Streams;
+
+namespace Orleans.Providers.Streams.PersistentStreams
+{
+    /// <summary>
+    /// Delivery failure table storage entity.
+    /// </summary>
+    public class StreamDeliveryFailureEntity : TableEntity
+    {
+        /// <summary>
+        /// Id of the subscription on which this delivery failure occured.
+        /// </summary>
+        public Guid SubscriptionId { get; set; }
+
+        /// <summary>
+        /// Name of the stream provider generating this failure.
+        /// </summary>
+        public string StreamProviderName { get; set; }
+
+        /// <summary>
+        /// Guid Id of the stream on which the failure occured.
+        /// </summary>
+        public Guid StreamGuid { get; set; }
+
+        /// <summary>
+        /// Namespace of the stream on which the failure occured.
+        /// </summary>
+        public string StreamNamespace { get; set; }
+
+        /// <summary>
+        /// Serialized sequence token of the event that failed delivery.
+        /// </summary>
+        public byte[] SequenceToken { get; set; }
+
+        /// <summary>
+        /// Sets the partition key before persist call.
+        /// </summary>
+        public virtual void SetPartitionKey(string deploymentId)
+        {
+            PartitionKey = String.Format("DeliveryFailure_{0}_{1}", StreamProviderName, deploymentId);
```

Comment:
These entities are only written when a stream provider fails, after multiple attempts, to deliver an event to a subscriber.  In the strict reliable case, this should be, by default, one failure per stream per minute per stream provider.  This is very low volume.  For other scenarios, which there may well be, the entity’s SetPartitionKey and SetRowKey are virtual, so more efficient keys, for write or read, can be generated.  I’m unconvinced we know how per formant this needs to be or what the query pattern is.  Given that, this pattern is a place to start with the flexibility to customize if/when issues surface.